CONTRIBUTING.md

```diff
@@ -1,77 +1,193 @@
+   Go through the [pull request template](#pull-request-template) in the pre-filled default description.
+
+7. Respond to review comments, potential CI failures and potential merge conflicts by updating the pull request.
+   Always keep the pull request in a mergeable state.
+
+   - To add new commits, repeat steps 3-4 and push the result using
+     ```
+     git push
+     ```
+
+   - To change existing commits you will have to [rewrite Git history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).
+     Useful Git commands that can help a lot with this are `git commit --patch --amend` and `git rebase --interactive`.
+     With a rewritten history you need to force-push the commits using
+     ```
+     git push --force-with-lease
+     ```
+
+   - In case of merge conflicts you will also have to [rebase the branch](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) on top of current `master`.
+     Sometimes this can be done [on GitHub directly](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch#updating-your-pull-request-branch), but if not you will have to rebase locally using
+     ```
+     git fetch upstream
+     git rebase upstream/master
+     git push --force-with-lease
+     ```
+
+   - If you need to change the base branch of the pull request, you can do so by [rebasing][rebase].
+
+8. If your pull request is merged and [acceptable for releases][release-acceptable] you may [backport][pr-backport] the pull request.
+
+### Pull request template
+[pr-template]: #pull-request-template
+
+The pull request template helps determine what steps have been made for a contribution so far, and will help guide maintainers on the status of a change. The motivation section of the PR should include any extra details the title does not address and link any existing issues related to the pull request.
+
+When a PR is created, it will be pre-populated with some checkboxes detailed below:
+
+#### Tested using sandboxing
+
+When sandbox builds are enabled, Nix will setup an isolated environment for each build process. It is used to remove further hidden dependencies set by the build environment to improve reproducibility. This includes access to the network during the build outside of `fetch*` functions and files outside the Nix store. Depending on the operating system access to other resources are blocked as well (ex. inter process communication is isolated on Linux); see [sandbox](https://nixos.org/nix/manual/#conf-sandbox) in Nix manual for details.
+
+Sandboxing is not enabled by default in Nix due to a small performance hit on each build. In pull requests for [nixpkgs](https://github.com/NixOS/nixpkgs/) people are asked to test builds with sandboxing enabled (see `Tested using sandboxing` in the pull request template) because in [Hydra](https://nixos.org/hydra/) sandboxing is also used.
```

Comment:
Is the performance hit the reason? I thought it was because the sandbox is not available on all platforms.

Reply:
This is just moved documentation, it was previously in https://nixos.org/manual/nixpkgs/stable/#submitting-changes-tested-with-sandbox, we shouldn't discuss changes to it here.